src/SprykerEco/Zed/Heidelpay/Business/Adapter/Basket/Basket.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+/**
+ * MIT License
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+namespace SprykerEco\Zed\Heidelpay\Business\Adapter\Basket;
+
+use Generated\Shared\Transfer\HeidelpayBasketRequestTransfer;
+use Generated\Shared\Transfer\HeidelpayBasketResponseTransfer;
+use Heidelpay\PhpBasketApi\Object\Authentication;
+use Heidelpay\PhpBasketApi\Object\Basket as HeidelpayBasket;
+use Heidelpay\PhpBasketApi\Request;
+use SprykerEco\Zed\Heidelpay\Business\Adapter\Mapper\BasketRequestToHeidelpayInterface;
+use SprykerEco\Zed\Heidelpay\Business\Adapter\Mapper\BasketResponseFromHeidelpayInterface;
+use SprykerEco\Zed\Heidelpay\HeidelpayConfigInterface;
+
+class Basket implements BasketInterface
+{
+    /**
+     * @var \SprykerEco\Zed\Heidelpay\HeidelpayConfigInterface
+     */
+    protected $config;
+
+    /**
+     * @var \SprykerEco\Zed\Heidelpay\Business\Adapter\Mapper\BasketResponseFromHeidelpayInterface
+     */
+    protected $responseMapper;
+
+    /**
+     * @var \SprykerEco\Zed\Heidelpay\Business\Adapter\Mapper\BasketRequestToHeidelpayInterface
+     */
+    protected $requestMapper;
+
+    /**
+     * @param \SprykerEco\Zed\Heidelpay\Business\Adapter\Mapper\BasketRequestToHeidelpayInterface $requestMapper
+     * @param \SprykerEco\Zed\Heidelpay\Business\Adapter\Mapper\BasketResponseFromHeidelpayInterface $responseMapper
+     * @param \SprykerEco\Zed\Heidelpay\HeidelpayConfigInterface $config
+     */
+    public function __construct(
+        BasketRequestToHeidelpayInterface $requestMapper,
+        BasketResponseFromHeidelpayInterface $responseMapper,
+        HeidelpayConfigInterface $config
+    ) {
+        $this->requestMapper = $requestMapper;
+        $this->responseMapper = $responseMapper;
+        $this->config = $config;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\HeidelpayBasketRequestTransfer $heidelpayBasketRequestTransfer
+     *
+     * @return \Generated\Shared\Transfer\HeidelpayBasketResponseTransfer
+     */
+    public function addNewBasket(HeidelpayBasketRequestTransfer $heidelpayBasketRequestTransfer): HeidelpayBasketResponseTransfer
```

Comment:
This method does 3 things in a row:
- builds request
- sends request
- maps request.
I would expect more separation here